### PR TITLE
Slimefun fireworks hurt you

### PR DESCRIPTION
**Pull request by Eoghanmc22**
## Description
When you research something in a closed space with fireworks turned on the fireworks hurt you.
## Changes
- Made slime fun fireworks not hurt you.
- Fixed exception on startup from f5eb90f

## Related Issues
none

## Testability

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.

It should work with add-ons as long as the add-ons aren't purposely spawning fireworks to hurt you.

I have never made a pull request before. Sorry if I messed up.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@
 *.iml
 /target
 /.idea/
-dependency-reduced-pom.xml
\ No newline at end of file
+dependency-reduced-pom.xml
+.DS_Store
```
Comment (WalshyDev) [inline .gitignore]: Add a slash at the start

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ResearchListener.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+public class ResearchListener implements Listener {
+	public ResearchListener(final SlimefunPlugin plugin) {
+		plugin.getServer().getPluginManager().registerEvents(this, plugin);
+	}
+
+	@EventHandler(priority = EventPriority.LOWEST)
+	public void onResearchFireworkDamage(final EntityDamageByEntityEvent e) {
+		if (e.getEntityType() == EntityType.PLAYER && e.getDamager() instanceof Firework)
+			if (((Firework) e.getDamager()).getFireworkMeta().getPersistentDataContainer()
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ResearchListener.java]: Create a Firework variable and clean this up a bit

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ResearchListener.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.persistence.PersistentDataType;
+
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+
+public class ResearchListener implements Listener {
+	public ResearchListener(final SlimefunPlugin plugin) {
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ResearchListener.java]: Empty line after class and before constructor

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ResearchListener.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+		plugin.getServer().getPluginManager().registerEvents(this, plugin);
+	}
+
+	@EventHandler(priority = EventPriority.LOWEST)
+	public void onResearchFireworkDamage(final EntityDamageByEntityEvent e) {
+		if (e.getEntityType() == EntityType.PLAYER && e.getDamager() instanceof Firework)
+			if (((Firework) e.getDamager()).getFireworkMeta().getPersistentDataContainer()
+					.get(new NamespacedKey(SlimefunPlugin.instance, "slime"), PersistentDataType.INTEGER) != null
+					&& ((Firework) e.getDamager()).getFireworkMeta().getPersistentDataContainer()
+							.get(new NamespacedKey(SlimefunPlugin.instance, "slime"), PersistentDataType.INTEGER) == 1)
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ResearchListener.java]: We have a util for persistent data containers (also, probably not really needed.... fireworks aren't exactly likely to be existing over restarts. I'd just add a metadata.

File: src/main/resources/languages/messages_ar.yml [hunk 5]
```diff
@@ -255,7 +255,7 @@ messages:
     whirlwind: "&a&oأبعدت تعويذتك القذيفة عنك"
     wizard: "&a&oحسنت تعويذتك حالة حظك لكن غالبا سوءت حالات أخرى"
   unknown-player: "&4لاعب غير معروف: &c%player%"
-  unlocked: &bتم فتح &7"%research%"
+  unlocked: '&bتم فتح&7"%research%"'
```
Comment (WalshyDev) [inline src/main/resources/languages/messages_ar.yml]: Looks like there's a fucked up speech mark though they all use it so I'd replace the speech mark with quoation marks to keep this consistent.

**Comment by TheBusyBiscuit** [comment]
Translations via GitHub are not allowed.

**Pull request closed without merging**